packages/fuels/tests/bindings.rs

```diff
@@ -801,3 +806,121 @@ async fn compile_bindings_enum_input() {
     let expected = "0000000021b2784f0000000000000000000000000000002a";
     assert_eq!(encoded, expected);
 }
+
+#[tokio::test]
+async fn shared_types() -> Result<(), Error> {
+    abigen!(
+            Contract(name="ContractA", abi="packages/fuels/tests/bindings/sharing_types/contract_a/out/debug/contract_a-abi.json"),
+            Contract(name="ContractB", abi="packages/fuels/tests/bindings/sharing_types/contract_b/out/debug/contract_b-abi.json"),
+        );
+    let wallet = launch_provider_and_get_wallet().await;
+
+    let deploy_contract = |a_or_b: &str| {
+        let string = format!(
+            "tests/bindings/sharing_types/contract_{a_or_b}/out/debug/contract_{a_or_b}.bin"
+        );
+        let wallet = &wallet;
+        async move {
+            Contract::deploy(
+                &string,
+                wallet,
+                TxParameters::default(),
+                StorageConfiguration::default(),
+            )
+            .await
+        }
+    };
```

Comment:
Why not use the `Deploy` command here and regroup everything?

Reply:
Great idea. Initially wasn't used since there were Scripts in this mix as well, and `setup_contract_test` setups only contracts.

 If we see a need for it, we can expand `setup_contract_test` to become something like `setup_test` and have scripts and predicates in it as well.
 
 Might not be a bad idea... might take me a few hours to implement. @digorithm ?

Reply:
I would rather do it in new issue

Reply:
Yes I don't think we need to do it now